documentation/DataLoader.md

```diff
@@ -0,0 +1,66 @@
+# CouchDB Data Loader
+
+(`scripts/deleteAndLoadSnapsets.sh`)
+
+This script is used to setup CouchDB database and is executed as a Kubernetes batch Job every time new version of the
+universal image is deployed to the cluster (also when cluster is initially created).
+
+It does following:
+
+- Converts the preferences in universal into `snapset` Prefs Safes and GPII Keys,
+- Optionally deletes existing database,
```

Comment:
Puzzled that the option here is to delete the entire database rather than merely all of the documents of type "snapset"

Reply:
There is a production test (`test:vagrantProduction`) that executes `vagrantCloudBasedContainers.sh` without its `--no-rebuild` flag.  When first run locally, or when run by CI, there is no database to delete, so deleting the database is irrelevant at that point.

But, if a developer runs the test a second time, they can at their option start from scratch with an empty database, or use the `--no-rebuild` flag and modify the database in situ, which is closer to a production environment.

The question is whether the first option -- start from scratch -- is useful.

Reply:
OK, I think this is just a documentation issue, since if I understand correctly, the action of steps 5 and 6 is actually the one of https://github.com/GPII/universal/blob/master/scripts/deleteAndLoadSnapsets.js - so we should just clarify the wording in the readme here to explain that steps 5 and 6 don't simply drop "snapsets and keys" but in particular only those keys which are associated with snapsets. I think it would be helpful for the comment here to explicitly link to or reproduce the comment at the head of the script https://github.com/GPII/universal/blob/master/scripts/deleteAndLoadSnapsets.js#L11 so that, for example, anyone invoking this script will do so in confidence that it will not delete user data (unless they enable GPII_CLEAR_INDEX, which should be supplied with a clear warning)

Reply:
OK, I've added to the README -- take a look.